geopandas/tools/hilbert_curve.py

```diff
@@ -0,0 +1,172 @@
+import pandas as pd
+import numpy as np
+
+
+def _hilbert_distance(gdf, total_bounds, p):
+    """
+    Calculate hilbert distance for a GeoDataFrame
+    int coordinates
+
+    Parameters
+    ----------
+    gdf : GeoDataFrame
+
+    total_bounds : Total bounds of geometries - array
+
+    p : The number of iterations used in constructing the Hilbert curve
+
+    Returns
+    ---------
+    Pandas Series containing hilbert distances
+
+    """
+    # Calculate bounds as numpy array
+    bounds = gdf.bounds.to_numpy()
+    # Calculate discrete coords based on total bounds and bounds
+    x, y = _continuous_to_discrete_coords(total_bounds, bounds, p)
+    # Calculate distance from morton curve
+    distances = _encode(p, x, y)
+
+    return pd.Series(distances, index=gdf.index, name="hilbert_distance")
+
+
+def _continuous_to_discrete_coords(total_bounds, bounds, p):
+
+    """
+    Calculates mid points & ranges of geoms and returns
+    as discrete coords
+
+    Parameters
+    ----------
+
+    total_bounds : Total bounds of geometries - array
+
+    bounds : Bounds of each geometry - array
+
+    p : The number of iterations used in constructing the Hilbert curve
+
+    Returns
+    ---------
+    Discrete two-dimensional numpy array
+    Two-dimensional array Array of hilbert distances for each geom
+    """
+
+    # Hilbert Side len
+    side_length = 2 ** p
+
+    # Calculate x and y range of total bound coords - returns array
+    xmin, ymin, xmax, ymax = total_bounds
+
+    # Calculate mid points for x and y bound coords - returns array
+    x_mids = (bounds[:, 0] + bounds[:, 2]) / 2.0
+    y_mids = (bounds[:, 1] + bounds[:, 3]) / 2.0
+
+    # Transform continuous int to discrete int for each dimension
```

Comment:
```suggestion
    # Transform continuous value to discrete integer for each dimension
```

(midpoints are floating point)